extensions/amp-analytics/0.1/vendors/subscriptions-propensity.js

```diff
@@ -19,19 +19,26 @@ import {jsonLiteral} from '../../../../src/json';
 const SUBSCRIPTIONS_PROPENSITY_CONFIG = jsonLiteral({
   'vars': {
     'clientId': 'CLIENT_ID(__gads)',
-    'data':
-      '{"skus": "${skus}","source": "${source}","active": "${active}","product": "${product}"}',
+    'startJson': '%7B',
```

Comment:
How are you sending the vars. If sent along with the request url, amp-analytics should be able to encode the value for you. Could you please double check if that's not the case. 

This would work, however the config is really difficult to read and not preferred.

Reply:
I agree, its horrible.  But they want it double encoded not single encoded and I didn't know how to make AMP double encode things unless I encoded it once myself and then it would auto-encode things the second time.

The other problem is that its bad for the publishers because they need to encode all their variables manually so that AMP can do the second encode on them.

Reply:
The double encoding is required because they stuffed 3 things into 1 thing.  I could try to convince them to change it but I suspect it's not high on their list.

Specifically, the query string goes like:
  ?event=encode(encode(publicationId):encode(eventName):encode(jsonObject))&otherValues=....
Then to read they:

   events=decode(queryString("event")).split(':')
   publicationId = decode(events[0])
   eventName = decode(events[1])
   jsonObject = parseJson(decode(events[1]))

Reply:
😨 😨 😨 
You're right that AMP doesn't allow one to encode variables. But if this is a common case, it would be trivial to add such support.

Reply:
I agree with you that asking publishers to encode all their variables is bad. It would be extremely difficult to debug with some errors : (